### DISCUSSION TITLE
Measures to totally block communication with Growatt

### DISCUSSION THREAD

**Discussion by tabascoz** [johanmeijer/grott#578] [General]
Hi There, starting this thread to ask feedback and collect ideas on how to totally block communication with Growatt. 

I have been turning my eyes away from the fact that my data has been shared outside of my control to Growatt and their limits of data transfer are not well documented. 

In the light of recent conversation and request from manufacturer ( see https://github.com/johanmeijer/grott/discussions/574 )  it has been stated that they're pretending to "fix" some things on their end. I suspect, as usual when manufacturers decide to be against Open Source Community, that they will or cloud implement changes just to prevent tools like Grott from working. 

With that, i notice : 

- Changes from Server IP being pushed by Growatt and breaking users setup (multiple issues opened) . 
- @peters59 comment in post (https://github.com/johanmeijer/grott/issues/557#issuecomment-2245012582) saying Shine-X stick firmware version 3.1.0.5 does not allow to enter fixed IP or set interval time 

I tried to totally block the comm setting it up a IP block on my broadband carrier router, but unfortunately this rendered Grott and the Shine sticks useless. I am not sure if this is related to DHCP issues with router's block  or if the sticks do talk with Growatt from time to time to keep them alive.  

I purchased a new router to put in between sticks and internet, will install Openwrt and do some sniffing to check that and make sure what's going on. 

Right now I have Grott / Grottserver set but have not implemented full block of Shine sticks to internet. 

Appreciate all your ideas, would not like to spend cycles with that but unfortunately and ultimately if Grott stops working I'll need to get rid of my three Growatt inverters and purchase another brand ( also do accept recommendations!)

**Comment by patcher-ms** [comment]
IP server changes aren’t necessarily malicious.
Assuming ShineWiFi uses `server.growatt.com` to locate the server, changing the IP address it points to isn’t particularly newsworthy.

When you blocked communications did you use Grott or GrottServer? Grott requires a server to be there since it just sits in the middle and processes the messages it sees. GrottServer should work.

Purchasing a new inverter seems overkill. Even if Growatt were to change their protocol in a way that breaks Grott there are ways to get the data directly from the inverter via modbus. Have a look at SolarAssistant but there are also other (more DIY) solutions

**Reply by tabascoz** [reply]
Yep, the changes aren't evil by themselves. But still they might be not exactly evil but to work in favor of Growatt interests. 

I have blocked it with grott and grottserver, but something has broken when i blocked it at FW level. I am unsure, for example, if ShineWifi could bypass the setup to go to localserver and talk with Growatt servers ,for example to receive fw updates. 

Gotta figure out why Grott and Shinewifi stopped reporting locally when i blocked outside traffic at the router. 

> Purchasing a new inverter seems overkill

Yep i know... but dont wanna spend efforts and time playing cat and mice game. Would need to think about modbus or changing Shinewifi stick firmware, i know there are alternatives, but i am not ruling out changing till i know i can have 100% blocked installation. Don't wanna be caught by surprise to learn i lost monitoring when I'm away given some crazy stuff from manufacturer.

**Comment by PedroKTFC** [comment]
I've been using grottserver (with a few mods) for months now and so am completely independent of Growatt. My system's mine. My data's mine. What I do with them is entirely for me.

**Reply by Tampa** [reply]
Please share more about this! I'm interested in cutting them off as well after they keep messing up with firmware updates.

**Reply by PedroKTFC** [reply]
I run both Grott and Grottserver. It's Grottserver that emulates the Growatt servers, receiving and sending messages to the data logger/inverter. I actually use my own forked version of grottserver as I do a lot of automatic changing of timeslots (those in the UK will be familiar with Octopus energy giving free power periods - when these happen I all but empty my battery by exporting power and then refilling it with free power). The current "official" version of grottserver doesn't have this capability.

**Comment by tabascoz** [comment]
My new router has arrived yesterday. I installed Openwrt on it. I was able to properly block ShineWifi sticks traffic to outside world and Grott is working fine. Something with the old broadband's router wasn't right and it was preventing Grott to operate, even though the firewall rule was clear to only block to Wan. 

At night I'll do a full system shutdown and restart to see if the sticks would properly initiate. If so, I understand the risk should be mitigated. Will report back.

**Reply by typxxi** [reply]
will become risky - cause in case of issues you are on your own.

a budy had thought the same, 3 or 4 years ago and in late summer 2022 he learned it he hard way.
proud offline user of a then rare or very expensive sph 4600 he faced an issue he thought a firmware update could and should fix.

Called growatt support, where the supporter was not happy to have no access and to discuss an errorand issues just what he was told, but could not see he first rejected a support with a firmware. argues went from each side to the other , but no solution, cause he was die hard cloudfree fan. 
next day, smart as he is, he called later in the evening hoping for a different support person. Luck was on his side, same story , no firmware update without proper diagnosis and access, but he was offered to ship the sph to frankfurt growatt service back then. 

For sure not, I want the most recent firmare and now and so on
Finally he got a stick with a new firmware and limitations which meant: your 2 devices(serials) only and the full risk on your side 

days later  I got  ebay notification , a growatt sph for sale, in our town, can not be true, even sph 4600 -  i need that
till i realised in the description "broken" and send a dm. He realized my name, so he called me and told me the story

first update went smoothly inverter came back and rruns fine, still the usual annoyances or as he called it: not worth the update

the  broken one did go the same way and then display and communication were stuck. first step; ac off, panels off, battery off and a long long break of 1 or 2 days. turned ac on - dead

7 or 10 days of endless support trials and of cause his intrnet treasures who promised to have found out how to he put it on ebay cause his lawyer had told him how stupid his usual behaviour is to show who he is (over 2 m tall and strong) even through a phone, cause he had no more warranty on this broken firmware failure or precisely: If  he could get the issue fixed by whomever, then the warranty would be back again, but not with a broken firmware issue where the first one worked. 

If he had done nothing, no issue, if he had conneted the holy sph to the internet incl. full cloud registration to let the growatt center do their firmware update job, then in case of the same outcome "broken firmware" Growatt would have fixed it and had to ix it. 

So be aware of that cause he had to pay peak corona prices for a 999 € sph 4600 in jan 2022  which meant in August where I also had to get another onem that he paid 2400€ or 2500 $

quite expensive and has hit his cloudfree nerve really hard . We do no longer speak about growatt sph inverter untill he has a serious issue he can not fix alone and also stopped  his believe in solutions that are easy and free with a 100% warranty, so he does no longer play and tinker with growatts. He is much calmer and calls and asks first.

I do not even know if growatt still ships usb sticks or media or sends firmware files out.

just to let you know the big impact it can have - our growatt warranty was 10 years plus 3 months and from a promotion 2 years plus = 12,25 years and 11 years were left on his machine

**Reply by Tampa** [reply]
@typxxi They cannot sell you a product with a 10 year warranty and then tell you the warranty is void based on a restriction they made up on the spot. Read their warranty agreement. Don't let them beat you over the head with nonsense like that. Consumers have rights and they can't make up rules as they go along. I'd speak to a lawyer if it gets into those levels of money as well, because this might be breaking all sorts of laws, let alone the question what their aim is with that in the first place.

**Reply by typxxi** [reply]
Sorry, I bet you are not german , no lawyer cause they have not changed the warranty. 
The customer has been unwilling to fullfill his duties which meant "ship it to us".
The customer has asked for a solution based on his proposal: Send me an USB Stick, I can do it on my own
They said no, and later yes, but only if he signed the paper that he would be doing everything at his own risk and that the damage  caused by his actions will not be covered by warranty.

He signed that paper and did the worst mistake at all in germany. He had been so stupid register him and products under a company name to get access to certain whatsoever. 
They used these details in thise paper regarding the risks and warranty and he signed it. 

And that alone changes everything. He had a lawyer and lawyers do earn more money from happy customers than from customers who have fought a big battle at court but lost. The lawyer is a great guy cause he delivered for me  what he had said in all cases before. And if things are not looking good, do not got to court and find a settlement, cause the amount and the hurt will be much less then having to pay the court, the others lawyer, maybe an court approven certified engineer and his lawyer for a big defeat for nothing, where usual lawyers promise: we fight that back and will win. 

I trust him, had a lot to do with insurances as car crash victim caused on the highway by someone else coming from the back too fast and under drugs. Lots at stake and long and winding battle I would not wann go throug again, but his poker game did work out by a surprising move to call the driver of the car who had caused the mess as whitness who then told the full truth to the surprise of the insurance volunteerly. My lawyer had visited him and told him that based on the speed he had been driving he would end in prison cause he would sit next to the general attorney. 
 
If you sign a letter and agree to cover the risk and then call for warranty for exact this what you had signed before then it is quite unusual at least from the root of modern law, cause law had started in europe, italy otherwise we would not have the famous quotes like "Nulla poena sine lege". 
 
Good luck without growatt cloud and support cause I have learned from roughly dozen inverters we have from them that without connection there is no support unless you ship the inverter to their repair facility. In all cases where they had no access and I had to describe the status of the inverter nothing had been achieved even after many hours. 

There is only one way to which has worked and that is a growatt manager from the near east and africa headquarter who had technicians and procedures to get such problems fix, but this has happened only twice in about 12 years with the same machine. But those were locals ,well trained domestic employees, our growatt supporters are more or less chineses expats with close to no german skills. 

have nice chrismas, have to pick up the chrismas tree cause we celebrate monday noon already and then 3 days.

**Reply by Tampa** [reply]
You cannot contractually sign away your rights or overrule law. That's the same everywhere, especially in Germany. I should know, I already had such fun incidents with other companies. Their tactic is to bully you to get you to sign things and what not, but unless you specifically waive your rights they can't deny a legal warranty.

I don't know what their whole business is with the data collection and why that's somehow so important to them, but it is concerning.

Again. Something like the whole John Deere thing in the US with contracts that prohibit a farmer from repairing their own tractor or seeking independent mechanics do so. That only worked there, because they didn't have laws against stuff like it. The EU, in all its nonsense, does take consumer protections a bit more serious. A company can't forbid you to do with your property as you see fit and retribution for doing so isn't allowed either. They can write what they want into their licenses and end user agreement, but that does not make those things hold up if they contain things that are against the law.

It's always difficult to deal with multi-national companies like this, but what generally helps the most is to remind them quite strongly that given what they are doing otherwise in terms of data collection and especially messing with grid-connected appliances isn't something they want to have come under scrutiny. The last thing they'd want is to mess up and cause problems for the energy providers or grid maintenance, because they won't hold a candle against those. They are one bad firmware and city-wide outage away from having the scope on them and I'm not so sure that everything they do would go over well with an energy company and accompanying government.

It's water under the bridge in that case, but that doesn't mean they are now allowed to do whatever they please. :)

**Reply by tabascoz** [reply]
Typxxi, thanks for sharing the story. I understand what you mean and the risks.  I am pro cloud-free person, however I wouldn't deny connecting the inverter back and letting Growatt investigate an issue. They have genuine rights to do so, it's all possible and better for them to diagnose an issue "in-loco". 

Not criticizing what the colleague has done, but I also agree with Tampa as they cannot go and cast a requirement out of the nowhere (like connecting to the cloud) and invalidate the warranty. Also because it is not uncommon for someone going off the grid that they might have true issues in being connected, like living in nowhere (Let's say middle of Africa or Brazil) without  internet connection. 

My suggestion to such cases would be erasing data logger and connecting to the grid just for the time they need access. With that, no historical records ever being sent, no privacy concerns. Risk is they update the firmware with something that could brake Grott (unlikely but possible). However, for what i need, it would be possible to use modbus and/or serial connection to control the hardware anyway. If that fails also i would end up selling and buying another inverters. 

I like Growatt, I truly think is a quality product. I have 4 SPFs in total, could see that they made several little enhancements to their hardware (and software) overtime. But it is concerning that they are assuming this posture. 

If all inverters brand starts acting like that and standing a non-friednly posture to OS community,  it wont take long for someone to build an OS inverter hardware and I'd bet China will start producing and selling it, like they do with many other OS hardware projects all around. 

What I am more attentive is that something like what happened to Deye in US might happen with Growatt inverters "off-cloud". But i think we are better served with EU-laws to protect our rights as customers.

**Comment by tabascoz** [comment]
So i think i have good news :) at least in sticks with  firmware 3.0.0.2, i can confirm even after full shutdown, the ShineWIFI sticks do properly initialize and work as expected, reporting to my Grott server. 

I am unsure what went wrong the first time i tried it, but i was already aiming to replace my broadband router anyway....
